llvm/test/CodeGen/X86/fold-pcmpeqd-2.ll

```diff
@@ -138,7 +141,10 @@ define void @program_1(ptr %dest, ptr %t0, <4 x float> %p0, <4 x float> %p1, <4
 ; X64-NEXT:    xorps %xmm0, %xmm0
 ; X64-NEXT:    movaps %xmm0, {{[-0-9]+}}(%r{{[sb]}}p) ## 16-byte Spill
 ; X64-NEXT:    movaps {{[-0-9]+}}(%r{{[sb]}}p), %xmm0 ## 16-byte Reload
-; X64-NEXT:    mulps %xmm0, %xmm0
+; X64-NEXT:    mulss {{\.?LCPI[0-9]+_[0-9]+}}(%rip), %xmm0
+; X64-NEXT:    movaps %xmm0, {{[-0-9]+}}(%r{{[sb]}}p) ## 16-byte Spill
+; X64-NEXT:    movaps {{[-0-9]+}}(%r{{[sb]}}p), %xmm0 ## 16-byte Reload
+; X64-NEXT:    shufps {{.*#+}} xmm0 = xmm0[0,0,0,0]
```

Comment:
NOTE: This regression is addressed by #74304

Reply:
If I read the IR correctly, it is a `zeroinitializer * zeroinitializer`. Why we still need a `mulss` here?

Reply:
Various hasOneUse limits prevents the constant data ever being seen as a fp constant, preventing constant folding.

Reply:
Its getting narrowed from mulps to mulss by scalarizeBinOpOfSplats